### PR TITLE
Create the combined Hue/Saturation control

### LINKED ISSUE #319: Create the combined Hue/Saturation control
## Description
Implement the HS control as a `UIControl` subclass and connect it to the state. Designs are here: https://toggl.invisionapp.com/d/main#/console/20014282/419193923/preview, while implementation specs can be found here: https://www.notion.so/toggl/Color-Picker-Unification-fe79a56649d74175a1b004936c7ebe10#1a1dc1f21c7f44f89adc95b666248458
TLDR: Hue changes from 0 to 360, and saturation changes for 30 to 100

## Dependencies
Depends on #318

## Definition of done
- [x] The HS control is implemented as a 2d control
- [x] The HS control parses the H and S values of the currently set custom colour and sets its thumb to the appropriate coordinates
- [x] The control's thumb is coloured into the currently set custom colour
- [x] The control's background changes depending on dark mode (see Notion doc)
- [x] The control's background gets lighter/darker depending on the currently set custom colour's V value
- [ ] Moving the control's thumb emits a .colorHueSaturationChanged action with a tuple containing the new Hue and Saturation pair. This action then emits a `.colorPicked` effect.

### PR DESCRIPTION
**Pull request by sofokli1**
### Summary
Creates the view and hooks it to the appropriate action in the VC

### Why is it done like this and not some other way?
Opted for drawing the whole view instead of using layer. Had some troubles with layers and this was easier.

### Relationships
Closes #319

### Pull Request checklist

- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [ ] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests

- [x] The code used custom templates

### Pointers
For testing the `value` changes you need to manually set it in code and run it again (until the slider is implemented)

### Review template
- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
🦑

### CODE REVIEW

File: Modules/Utils/Utils/Extensions/Comparable+Extention.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+public extension Comparable {
+    func clamp(min: Self, max: Self) -> Self {
```
Comment (juanlaube) [inline Modules/Utils/Utils/Extensions/Comparable+Extention.swift:4]: Amazing! 😄

File: Modules/Timer/Timer/Project/Views/HueSaturationPickerView.swift [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+class HueSaturationPickerView: UIControl {
+    // swiftlint:disable no_ui_colors
+    private let circleDiameter: CGFloat = 30
+    private var circleRadius: CGFloat { circleDiameter / 2 }
+
+    private let outerCircleDiameter: CGFloat = 32
+    private var outerCircleRadius: CGFloat { outerCircleDiameter / 2 }
+    private let outerCircleColor = UIColor.white.projectColor
+
+    private static let defautlColor = UIColor(hex: "3178be").projectColor
```
Comment (juanlaube) [inline Modules/Timer/Timer/Project/Views/HueSaturationPickerView.swift:13]:
```suggestion
    private static let defaultColor = UIColor(hex: "3178be").projectColor
```

**Comment by sofokli1** [comment]
@juanlaube the same thing happens in the old app. @dshr can you clarify how it should behave?

File: Modules/Timer/Timer/Project/Views/HueSaturationPickerView.swift [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        drawOuterCircleWithShadow(rect: outerCircleRect)
+
+        let circleRect = CGRect(x: x, y: y, width: circleDiameter, height: circleDiameter)
+        drawColorCircle(rect: circleRect)
+    }
+
+    private func drawRainbowBackground(rect: CGRect) {
+        let context = UIGraphicsGetCurrentContext()!
+
+        let red = UIColor.red.projectColor
```
Comment (dshr) [inline Modules/Timer/Timer/Project/Views/HueSaturationPickerView.swift:84]: For some reason these don't change when switching between light/dark mode.

File: Modules/Timer/Timer/Project/Views/HueSaturationPickerView.swift [hunk 4]
```diff
@@ -81,13 +81,13 @@ class HueSaturationPickerView: UIControl {
...
         let context = UIGraphicsGetCurrentContext()!
 
-        let red = UIColor.red.projectColor
-        let yellow = UIColor.yellow.projectColor
-        let green = UIColor.green.projectColor
-        let cyan = UIColor.cyan.projectColor
-        let blue = UIColor.blue.projectColor
-        let magenta = UIColor.magenta.projectColor
-        let borderColor = UIColor(hex: "CECECE").projectColor
+        let red = UIColor.red
```
Comment (dshr) [inline Modules/Timer/Timer/Project/Views/HueSaturationPickerView.swift:101]: The gradient colours should have the dark mode transformations applied to them. My previous comment meant that the code looked okay, and that the transforms should've worked, but they didn't for some reason, and I guess I should've phrased it better, sorry about that.

**Pull request merged**